controllers/teleport_master_controller/teleport_master_controller.cpp

```diff
@@ -0,0 +1,144 @@
+/*
+ * This file is part of the NUbots Codebase.
+ *
+ * The NUbots Codebase is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The NUbots Codebase is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with the NUbots Codebase.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright 2021 NUbots <nubots@nubots.net>
+ */
+ 
+// Special thanks to Hamburg Bit-Bots for ideas for this controller
+// https://github.com/bit-bots/wolfgang_robot/blob/feature/recognition/wolfgang_webots_sim/src/wolfgang_webots_sim/webots_camera_controller.py
+
+#include <array>
+#include <fstream>
+#include <iostream>
+#include <math.h>
```

Comment:
```suggestion
#include <cmath>
```

Reply:
What's the benefit of using this library over math.h?

Reply:
`math.h` is the C header, `<cmath>` is the c++ header and we are using c++

Technically according to the c++ standard, you would have been expected to wrap your `<math.h>` in an extern "C" block but basically every standard library does that for you in macros.

Reply:
I see I see do they not provide the same functionality?

Reply:
For the most part, yes. The C++ library does more though. It provides overloading so the same function can be used with different types (C can't do this at all, instead they have something like `sqrt` for double and `sqrtf` for floats)
```c++
float sqrt(float f);
double sqrt(double f);
```

Probably does more, but most if it is around overloading and better type support

Reply:
Yeah okay gotcha. Thanks :)

Reply:
Yeah if you include "math.h" it often doesn't include the std:: versions of the functions (although sometimes it does).
This is where keeping to the standard rather than "what works" is important in c++.
If you just go with whatever works you can end up trying to compile your code on another compiler (clang vs gcc vs msvc) and it'll sometimes work and sometimes not which can drive you insane.

If you stick to the standard then it'll work everywhere (unless there is a bug in the compiler :P)